src/backend/distributed/planner/recursive_planning.c

```diff
@@ -553,6 +1229,39 @@ ShouldRecursivelyPlanSubquery(Query *subquery)
 }
 
 
+/*
+ * AllDistributionKeysInSubqueryAreEqual is a wrapper function
+ * for AllDistributionKeysInQueryAreEqual(). Here, we filter the
+ * planner restrictions for the given subquery and do the restriction
+ * equality checks on the filtered restriction.
+ */
+static bool
+AllDistributionKeysInSubqueryAreEqual(Query *subquery,
+									  PlannerRestrictionContext *restrictionContext)
+{
+	bool allDistributionKeysInSubqueryAreEqual = false;
+	PlannerRestrictionContext *filteredRestrictionContext = NULL;
+
+	/* we don't support distribution eq. checks for CTEs yet */
+	if (subquery->cteList != NIL)
```

Comment:
Not super important, but I suppose you can make an exception for CTEs without distributed tables.

Reply:
We don't support eq. of local tables anyway. So, keeping it for now as it is?

Reply:
I was thinking more in terms of constant CTEs and reference tables, but we can skip this.